### PR TITLE
#12 rename property

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
keep details displayed on change
improve name check (as python identifier)
hide Auth
fold pipelines and scenarios treeviews
v 0.4

### CODE REVIEW

File: l10n/bundle.l10n.json [hunk 1]
```diff
@@ -31,6 +30,9 @@
   "Main module file has been set up as {0} in Workspace settings": "Main module file has been set up as {0} in Workspace settings",
   "Create a new function": "Create a new function",
   "Create a new class": "Create a new class",
+  "Not a valid Python identifier.": "Not a valid Python identifier.",
+  "Entity {0} Name should be a valid python identifier and not 'default': '{1}'": "Entity {0} Name should be a valid python identifier and not 'default': '{1}'",
```
Comment (FabienLelaquais) [inline l10n/bundle.l10n.json]: I think 'Entity' is unfortunate here. That's a specific Core concept.

'python' must be capitalized.

File: l10n/bundle.l10n.json [hunk 2]
```diff
@@ -31,6 +30,9 @@
...
+  "Another {0} entity has the name {1}": "Another {0} entity has the name {1}",
```
Comment (FabienLelaquais) [inline l10n/bundle.l10n.json]: Same remark here.
'element'?

File: src/context.ts [hunk 3]
```diff
@@ -239,16 +246,24 @@ export class Context {
...
+  updateSelectionCache(nodeType: string, oldNodeName: string, nodeName: string) {
+    if (this.selectionCache[nodeType] === oldNodeName) {
+      this.selectionCache[nodeType] = nodeName;
+      this.vsContext.workspaceState.update(Context.cacheName, this.selectionCache);
+    }
+  }
+
   private async selectConfigNode(nodeType: string, name: string, configNode: object, uri: Uri, reveal = true, fromInEditor = true): Promise<void> {
     let updateCache = false;
+    console.log(`In selectConfigNode(nodeType: ${nodeType}, name: ${name}, configNode: object, uri: Uri, reveal = ${reveal}, fromInEditor = ${fromInEditor}) cache: ${this.selectionCache.lastView}`);
```
Comment (FabienLelaquais) [inline src/context.ts]: Sure about that?

File: src/utils/pythonSymbols.ts [hunk 4]
```diff
@@ -55,4 +55,20 @@ export const getModulesAndSymbols = async (isFunction: boolean): Promise<[string
...
+};
+
+const IDENTIFIER_RE = /^[A-Za-z]\w*$/;
+const isValidPythonIdentifier = (value: string) => !!value && IDENTIFIER_RE.test(value);
+export const checkPythonIdentifierValidity = (value: string) => isValidPythonIdentifier(value) ? null: l10n.t("Not a valid Python identifier.");
+
+export const getNodeNameValidationFunction = (typeSymbol?: DocumentSymbol, nodeName?: string) => {
+  return (value: string) => {
+    if (!isValidPythonIdentifier(value) || value.toLowerCase() === "default") {
+      return l10n.t("Entity {0} Name should be a valid python identifier and not 'default': '{1}'", typeSymbol?.name, value);
```
Comment (FabienLelaquais) [inline src/utils/pythonSymbols.ts]: Check this

File: src/utils/pythonSymbols.ts [hunk 5]
```diff
@@ -55,4 +55,20 @@ export const getModulesAndSymbols = async (isFunction: boolean): Promise<[string
...
+    }
+    if (value !== nodeName && typeSymbol?.children.some(s => s.name === value)) {
+      return l10n.t("Another {0} entity has the name {1}", typeSymbol?.name, value);
```
Comment (FabienLelaquais) [inline src/utils/pythonSymbols.ts]: ... and this

**Review by FabienLelaquais — APPROVED** [review]
Nice piece of work.
A few mandatory changes though.

File: src/context.ts (hunk 3 above)
Comment (FredLL-Avaiga) [reply src/context.ts]: good catch

**Pull request merged**